.github/workflows/ci.yml

```diff
@@ -35,20 +35,21 @@ jobs:
     - name: Check ruff
       run: |
         source .venv/bin/activate
-        ruff check .
-        ruff format --check .
+        cd python
+        ruff check
+        ruff format --check
     - name: Test with pytest and report coverage
       run: |
         source .venv/bin/activate
-        cd tests
+        cd python/tests
         coverage run --branch --source=../ -m pytest
         coverage report
-        cd ..
```

Comment:
Oh nice, does gh actions do some cleanup between steps? This was actually always a pain in circleci

Reply:
Yeah I looked into this, seems like each step starts a fresh shell from the project root, while the filesystem is shared across all steps.